### ISSUE TITLE
In fast group muter, the value cannot be input

### ISSUE THREAD

**Issue by MartinTremblay** [rgthree/rgthree-comfy#214]
right now, we cannot turn the widget into input. 

It would be useful to have it to 'automatically' turn on or off a group using the output of another node.

**Comment by saftle** [comment]
100%. I'd absolutely love to have this. It's the only thing missing from my workflow, since I need to pass a 0 or 1 value to disable the specific groups if possible externally.

**Comment by rgthree** [comment]
This is hard to explain so apologies if it doesn't make sense.

But there are essentially two types of nodes/features in ComfyUI: "client-side" ones that can manipulate the UI and the workflow you're looking at in the browser, and then "server-side" nodes that run on the server while the workflow is being executed, _on the server_.

Fast Groups Muter is client-side (in fact, all muting/bypassing in ComfyUI is client-side). It works only in the UI itself and responds to clicks on the its toggle at that click-moment to mute/bypass/enable groups in the current workflow you're working in. 

When you send your workflow to the server, the server goes node by node following the connections, evaluates each node, and then moves to the next node using the previous nodes' output values for the connected inputs.

What this means, unfortunately, is that inputs don't actually have values on the client side; they just represent connections that the server will evaluate _after_ the workflow is sent to the server. Further, on the server while nodes are being evaluated, you can't mute/bypass/enable nodes as we've already sent the workflow to the server where it's currently running.

Basically, an input on "Fast Groups Muter / Bypasser" won't know what the value of an input is until its too late to try to mute/bypass something on the client-side.

**Comment by rgthree** [comment]
I don't think this is what's being asked, but if your goal is to change the mode of a group based on another groups' mode, that can be done using Relay/Repeater:

![Screenshot 2024-05-04 215715](https://github.com/rgthree/rgthree-comfy/assets/476360/577339f6-75a2-4966-8c84-4b1556f5e27c)

In the screenshot, Group B will be automatically muted/bypassed/enabled when Group A is muted/bypassed/enabled.

You can also change Relay to change the outgoing mode. So you could build flows like "Group B should be enabled when Group A is muted" etc.

Again, I don't think this is what the OP is asking, but figured it's worth mentioning here.

**Comment by MartinTremblay** [comment]
In my specific case, I have a custom node that run prompt from an external json list and I want to run a face detailer workflow if the prompt contains specific words. I ended up using the "Flood gate" custom node to do that. In the end, I just needed a boolean output from another node as the input for that logic node.

I was the first time a saw a node without the ability to switch from widget to input.

**Comment by rgthree** [comment]
FloodGate is a good node, especially if it works for your case. However, it does fall victim to the same thing I described above with not being able to use the input value from an executed output.

In it's own docs:

> Note: The logic flow is parsed during the queuing stage, not the execution stage. As a result, the boolean value has to be already determined when you press Queue Prompt, such as from the toggle or a primitive node. If you use a node that outputs a boolean during execution, this will not work.

(Here, "queue stage" is like "client side" I used above, and "execution stage" is like "server side" I used above).

Given these docs, it looks like it works the same way as the fast muter/bypasser would; that you have to manually toggle it open or closed.

If you're using a node as an input (that isn't just a toggle/primitive node) I am confused how that works, especially based on something like prompt text. Which node connects to the floodgate's gate_open ?

**Comment by MartinTremblay** [comment]
I get it. I never thought it was even possible to "calculate" a value during processing.

In my case, I have an external file that have all the input value that need to change (prompt, lora, seed, width, height, output filename, etc.) and one of those output value is whether I need to run the face detailer or not. So basically the flood gate will either save the image or go thru the face detailer before saving.

Using a node to automatically turn on or off a group made more sense to me (it still do), but I managed to do what it needed it to do. Just wish I discovered that node few month ago.
